docs/source/cpp/parquet.rst

```diff
@@ -32,6 +32,298 @@ is a space-efficient columnar storage format for complex data.  The Parquet
 C++ implementation is part of the Apache Arrow project and benefits
 from tight integration with the Arrow C++ classes and facilities.
 
+Reading Parquet files
+=====================
+
+The :class:`arrow::FileReader` class reads data for an entire
+file or row group into an :class:`::arrow::Table`.
+
+The :class:`StreamReader` and :class:`StreamWriter` classes allow for
+data to be written using a C++ input/output streams approach to
+read/write fields column by column and row by row.  This approach is
```

Comment:
Is this value by value, or a choice between reading full rows or full columns?

Reply:
value by value, IIUC. I didn't touch this part of this docs since this part of the interface isn't that important.